intro-dplyr.Rmd

```diff
@@ -0,0 +1,343 @@
+    - Spread the column `type` into two columns `cases` and `population`.
+    
+```{r}
+table2 %>%
+    spread(key = type, value = count)
+```
+
+## Case study for using tidyr
+
+The `tidyr::who` dataset contains tuberculosis (TB) cases broken down by year, country, age, gender, and diagnosis method.
+
+Explore `who` data: are they tidy?
+
+```{r}
+glimpse(who)
+```
+
+`new_sp_m014` to `newrel_f65` are likely values, not variables. We don't know what those values represent yet, so we'll give them the generic name "key". We know the cells represent the count of cases, so we'll use the variable cases. Let's reshape these data:
+
+```{r}
+who1 <- who %>% 
+  gather(new_sp_m014:newrel_f65, key = "key", value = "cases", na.rm = TRUE)
+who1
+```
+
+The first three letters of each column denote whether the column contains new or old cases of TB. In this dataset, each column contains new cases.
+
+The next two letters describe the type of TB:
+
+- rel = relapse
+- ep = extrapulmonary TB
+- sn = smear negative
+- sp = smear positive
+The sixth letter gives the sex of TB patients. The dataset groups cases by males (m) and females (f).
+
+The remaining numbers gives the age group. The dataset groups cases into seven age groups:
+
+- 014 = 0 - 14 years old
+- 1524 = 15 - 24 years old
+- 2534 = 25 - 34 years old
+- 3544 = 35 - 44 years old
+- 4554 = 45 - 54 years old
+- 5564 = 55 - 64 years old
+- 65 = 65 or older
+
+Fixing inconsistent column names (`new_sp_` but `newrel_`)
+
+```{r}
+who2 <- who1 %>% 
+  mutate(key = gsub(pattern = "newrel", replacement = "new_rel", x = key))
+who2
+```
+
+Fixing an issue of having several values (e.g., description of the case, type of TB, sex, and age) are coded as one value:
+
+```{r}
+who3 <- who2 %>% 
+  separate(key, c("new", "type", "sexage"), sep = "_")
+who3
+```
+
+Dropping redundant columns `new`, `iso2`, `iso3`:
+
+```{r}
+who4 <- who3 %>% 
+  select(-new, -iso2, -iso3)
+str(who4)
+```
+
+Separating `sexage` values:
+
+```{r}
+who5 <- who4 %>% 
+  separate(sexage, c("sex", "age"), sep = 1)
+who5
+```
+
+### Exercise: Separate a column into two new ones
+
+Time: 10 min
+
+```{r}
+# explore the structure of the table3
+table3
+
+# separate rate variable into two columns with numeric values.
+table3 %>%
+  separate(col = rate, into = c("cases", "population"), sep = "/", convert = T) %>%
+  # calculate new variable "rate" per 100,000 population for each country per year
+  mutate(
+    rate = as.numeric(cases / population * 100000)
+  )
+```
+
+
+## Final exercise:
+
+Time: Until end of the session
+
+1. Choose a dataset to work with.
+2. Create a new variable using `mutate()`.
+3. Select variables relevant for the future summary using `select()`.
+4. Filter data based on several row values using `filter()`.
+5. Group your data before summarising if necessary using `group_by()`.
+6. Create a concise summary of the data using `summarise()`.
+7. Arrange results of your summary using `arrange()`.
```

Comment:
Instead of a series of steps, maybe have them try to recreate an output that's been wrangled. I can try to create an example.